pandas/tests/indexes/datetimes/test_datetime.py

```diff
@@ -17,6 +17,15 @@
 
 
 class TestDatetimeIndex:
+    def test_sub_datetime_preserves_reso(self, tz_naive_fixture):
```

Comment:
Could you add a test where this goes over a DST transition?

Reply:
good catch.  when we cross a DST transition, dti.freq is actually wrong bc of the (deprecated) kludge in generate_range.  As a result, inheriting the freq is incorrect in this case.  But it will return to being correct once the deprecation is enforced.

i think the thing to do is add the test and xfail it

Reply:
test added (and xfailed) + green